lib/default.nix

```diff
@@ -148,31 +148,141 @@ let
           _module.args.perSystem = systemArgs.${pkgs.system}.perSystem;
         };
 
+      # Sets up declared users without any user intervention, and sets the
+      # options that most people would set anyway. The module is only returned
+      # if home-manager is an input and the host has at least one user with a
+      # home manager configuration. With this module, most users will not need
+      # to manually configure Home Manager at all.
+      mkHomeUsersModule =
+        hostname: homeManagerModule:
+        let
+          module =
+            { perSystem, ... }:
+            {
+              imports = [ homeManagerModule ];
+              home-manager.sharedModules = [ perSystemModule ];
+              home-manager.extraSpecialArgs = specialArgs;
+              home-manager.users = homesNested.${hostname};
+              home-manager.useGlobalPkgs = lib.mkDefault true;
+              home-manager.useUserPackages = lib.mkDefault true;
+            };
+        in
+        lib.optional ((inputs ? home-manager) && (builtins.hasAttr hostname homesNested)) module;
```

Comment:
Since blueprint is hard-coding the path patterns, let's also explicitly throw an error if the required input is missing: "hey, you put a file in XY location so we need the home-manager input.".

At some point I want to make the lib/default.nix extensible, but we're not quite there yet.

Reply:
Good point. There are a couple ways this can be done:
* Throw when evaluating the homeManagerConfiguration
* Throw when evaluating the users directories
* Throw at some other point

There are advantages and disadvantages to each of these. I ended up going with option 2, throwing when evaluating the users directories. This is much more elegant, but it does mean that the error is somewhat 'eager' in that it will still be thrown even if not building a for system home-manager is needed on. That actually may make it the most correct option though, as it would be an error to define a home configuration without home manager even if it's not the system you're building